stl/inc/xutility

```diff
@@ -4491,6 +4506,18 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+template <class _Ty1, class _Ty2, class = void>
+_INLINE_VAR constexpr bool _Can_compare_with_operator_equal = false;
+
+template <class _Ty1, class _Ty2>
+_INLINE_VAR constexpr bool
+    _Can_compare_with_operator_equal<_Ty1, _Ty2, void_t<decltype(_STD declval<_Ty1&>() == _STD declval<_Ty2&>())>> =
+        true;
+
+template <class _Ty1, class _Ty2>
+_INLINE_VAR constexpr bool _Is_pointer_address_comparable =
+    _Can_compare_with_operator_equal<_Ty1*,
+        _Ty2*> && (_Is_pointer_address_convertible<_Ty1, _Ty2> || _Is_pointer_address_convertible<_Ty2, _Ty1>);

```

Comment:
This can fail in the presence of multiple inheritance since a pointer-to-base and pointer-to-derived can be equal but have differing bit patterns (https://godbolt.org/z/hcT9c9615).

Reply:
I believe that the use of `is_pointer_interconvertible_base_of` handles this scenario - but we should make sure that there's test coverage for this scenario (I haven't checked yet).

Reply:
Aha! I misread `_Is_pointer_address_convertible` completely.